Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Writers/YamlWriter.cs

```diff
@@ -0,0 +1,241 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using YamlDotNet.RepresentationModel;
+using YamlDotNet.Serialization;
+
+namespace Amazon.Lambda.Annotations.SourceGenerator.Writers
+{
+    /// <summary>
+    /// This contains the functionality to surgically manipulate a YAML blob
+    /// </summary>
+    public class YamlWriter
+    {
+        private YamlNode _rootNode;
+
+        public YamlWriter()
+        {
+            _rootNode = new YamlMappingNode();
+        }
+
+        public YamlWriter(YamlMappingNode rootNode)
+        {
+            _rootNode = rootNode;
+        }
+
+        /// <summary>
+        /// Checks if the dot(.) seperated yamlPath exists in the YAML blob stored at the _rootNode
+        /// </summary>
+        /// <param name="yamlPath">dot(.) seperated path. Example "Person.Name.FirstName"</param>
+        /// <returns>true if the path exist, else false</returns>
+        /// <exception cref="InvalidDataException">Thrown if the yamlPath is invalid</exception>
+        public bool Exists(string yamlPath)
+        {
+            if (!IsValidPath(yamlPath))
+            {
+                throw new InvalidDataException($"'{yamlPath}' is not a valid {nameof(yamlPath)}");
+            }
+
+            var currentNode = _rootNode;
+            foreach (var property in yamlPath.Split('.'))
+            {
+                try
```

Comment:
can you please add a null check for currentNode?

Reply:
Done